### PR TITLE
runtime: add vmstate::Registers unit tests

### PR DESCRIPTION
**Pull request by mina86**

### CODE REVIEW

**Comment by akhi3030** [comment]
@Ekleog-NEAR: can you review this PR please?  I will approve it once you sign off on it.

File: runtime/near-vm-logic/src/vmstate.rs
Comment (nagisa) [inline runtime/near-vm-logic/src/vmstate.rs:224]: I think this code would become slightly more readable if the methods largely concerned with asserting contained `assert_` or `check_` in them. That is `assert_read` or `check_read`, `assert_set_fails`, `assert_gas`.

Not a huge deal, though, since this is pretty self-contained.

**Review by nagisa — APPROVED** [review]

**Pull request merged**
